### DISCUSSION TITLE
R with conda in nextflow

### DISCUSSION THREAD

**Discussion by hkaspersen** [nextflow-io/nextflow#3265] [Q&A] [answered]
For a report process, I am using a docker image, with R, that has been manually created and uploaded online to dockerhub. This R image contains specific R library versions in addition to R itself.
However, I want my pipeline to be able to use conda as well as docker/singularity. My question is then: what is best practices here? Do I have to define a conda environment `yml` file and refer to it in the process definition, or how is it done?

My processes have the following defined for containers and conda (example from another process than the R process)
```
	conda (params.enable_conda ? 'bioconda::seqkit=2.1.0' : null)
	container 'quay.io/biocontainers/seqkit:2.1.0--h9ee0642_0'
```

This works fine for processes with only one program, but I do not know how to define this for a conda env with R + specific package versions.

**Comment by jemunro** [comment, marked as answer]
In this case it would probably be better to use a conda environment file: https://www.nextflow.io/docs/latest/conda.html#use-conda-environment-files

**Reply by hkaspersen** [reply]
This only shows how bad I am to read the documentation! Thanks!
